pkg/broker/awsbroker.go

```diff
@@ -241,127 +211,130 @@ func (b *AwsBroker) ValidateBrokerAPIVersion(version string) error {
-					for pk, pv := range param.(map[interface{}]interface{}) {
-						switch pk {
-						case "name":
-							name = pv.(string)
-						case "required":
-							required = pv.(bool)
-						case "type":
-							switch pv {
-							case "enum":
-								pvals[pk.(string)] = "string"
-							case "int":
-								pvals[pk.(string)] = "integer"
-							default:
-								pvals[pk.(string)] = pv
-							}
-						case "updatable":
-							updatable = pv.(bool)
-						default:
-							pvals[pk.(string)] = pv
-						}
-					}
-					propsForCreate[name] = pvals
-					if required {
-						requiredForCreate = append(requiredForCreate, name)
-					}
-					if updatable {
-						propsForUpdate[name] = pvals
-						if required {
-							requiredForUpdate = append(requiredForUpdate, name)
-						}
-					}
-				}
-				plan.Schemas = &osb.Schemas{
-					ServiceInstance: &osb.ServiceInstanceSchema{
-						Create: &osb.InputParametersSchema{
-							Parameters: map[string]interface{}{
-								"type":       "object",
-								"properties": propsForCreate,
-								"$schema":    "http://json-schema.org/draft-06/schema#",
-							},
-						},
-					},
+	params := cfnParamsToOsb(sd)
+	for k, p := range sd.Metadata.Spec.ServicePlans {
+		planid := uuid.NewV5(db.Accountuuid, "service__"+sd.Metadata.Spec.Name+"__plan__"+k).String()
+		plan := osb.Plan{
+			ID:          planid,
+			Name:        k,
+			Description: p.Description,
+			Free:        &f,
+			Bindable:    &t,
+			Metadata: map[string]interface{}{
+				"cost":            p.Cost,
+				"displayName":     p.DisplayName,
+				"longDescription": p.LongDescription,
+			},
+			Schemas: &osb.Schemas{ServiceInstance: &osb.ServiceInstanceSchema{}},
+		}
+		propsForCreate := make(map[string]interface{})
+		for nk, nv := range nonCfnParamDefs {
+			propsForCreate[nk] = nv
+		}
+		propsForUpdate := make(map[string]interface{})
+		requiredForCreate := make([]string, 0)
+		requiredForUpdate := make([]string, 0)
+		prescribed := make(map[string]string)
+		var openshiftFormCreate []OpenshiftFormDefinition
+		var openshiftFormUpdate []OpenshiftFormDefinition
+		for paramName, paramValue := range params {
+			include := true
+			for planParam, planValue := range p.ParameterValues {
+				if planParam == paramName {
+					include = false
+					prescribed[planParam] = planValue
 				}
-				if len(requiredForCreate) > 0 {
-					// Cloud Foundry does not allow "required" to be an empty slice
-					plan.Schemas.ServiceInstance.Create.Parameters.(map[string]interface{})["required"] = requiredForCreate
+			}
+			required := false
+			if paramValue.(map[string]interface{})["required"] != nil {
+				required = paramValue.(map[string]interface{})["required"].(bool)
+			}
+			if include {
+				openshiftFormCreate = openshiftFormAppend(openshiftFormCreate, paramName, paramValue.(map[string]interface{}))
+				createParam := paramValue.(map[string]interface{})
+				delete(createParam, "required")
+				delete(createParam, "display_group")
```

Comment:
We're removing `display_group` from these parameters but not from `nonCfnParamDefs`. Should we be removing it from both?